torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -196,8 +196,24 @@ def generate_ttir(
 
     assert isinstance(kernel, JITFunction)
 
+    context = triton._C.libtriton.ir.context()
+    target = triton.runtime.driver.active.get_current_target()
+    backend = triton.compiler.compiler.make_backend(target)
+    options = backend.parse_options({})
+
+    # ignore backend-specific kwargs same way as in the native Triton code
+    # https://github.com/triton-lang/triton/blob/a6bb57d6285e723c58e87dd7cba263db6efff789/python/triton/runtime/jit.py#L594-L596
+    # why this is important for user-defined Triton kernels on AMD: https://github.com/pytorch/pytorch/issues/140800
+    excess_kwargs = {name for name in kwargs if name not in kernel.arg_names}
+    for name in excess_kwargs:
+        if name in options.__dict__:
+            kwargs.pop(name)
```

Comment:
```suggestion
    for name in kwargs:
        if name not in kernel.arg_names and name in options.__dict__:
            kwargs.pop(name)
```

Reply:
@oulgen Updated, except `for name in list(kwargs)`, as can't modify the dict while iterating (getting "RuntimeError: dictionary changed size during iteration"). PTAL, thanks!